pypesto/store/auto.py

```diff
@@ -23,6 +24,9 @@ def autosave(
         Either the filename to save to or "Auto", in which case it will
         automatically generate a file named
         `year_month_day_{type}_result.hdf5`.
+        A method can also be provided. All input to the autosave method will
+        be passed to the filename method. The output should be the filename
+        (`str`).
     result:
```

Comment:
This would then only allow for result, store_type and overwrite to be passed to the callable? perhaps also add *kwargs there?

Reply:
Could you describe the use case? i.e. what else can be passed from `autosave` to `default_filename`?